keras_nlp/layers/transformer_decoder.py

```diff
@@ -166,7 +166,7 @@ def _feed_forward(self, input):
     def call(
         self,
         decoder_sequence,
-        encoder_sequence,
+        encoder_sequence=None,
```

Comment:
We might want to note in the docstring that for decoder-only models like GPT-2, `encoder_sequence` should be left as None.

Reply:
Also rethinking about this - we probably want to change the constructor as well to have an extra argument `decoder_only`, and when `decoder_only=True`, we don't need to create the layers related to encoder sequence.